### PR TITLE
Implement ControlledResources

### PR DESCRIPTION
**Pull request by tweks**
This implements functionality which allows specifying resources for which VPA generates recommendations.

API changes: https://github.com/kubernetes/autoscaler/commit/2cfdb76281bed5bcd4afccfde91ba379dbb9ab10

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/logic/recommender_test.go [hunk 1]
```diff
@@ -63,3 +63,29 @@ func TestMinResourcesSplitAcrossContainers(t *testing.T) {
 	assert.Equal(t, model.MemoryAmountFromBytes((*podMinMemoryMb*1024*1024)/2), recommendedResources["container-2"].Target[model.ResourceMemory])
 	assert.Equal(t, model.MemoryAmountFromBytes((*podMinMemoryMb*1024*1024)/2), recommendedResources["container-2"].Target[model.ResourceMemory])
 }
+
+func TestControlledResourcesFiltered(t *testing.T) {
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/logic/recommender_test.go:67]: Can we have one more test for [CPU, memory]? To make sure default works as expected.

File: vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state_test.go [hunk 2]
```diff
@@ -247,3 +247,46 @@ func TestUpdateFromPolicy(t *testing.T) {
...
+			policy: &vpa_types.ContainerResourcePolicy{
+				ControlledResources: &[]apiv1.ResourceName{apiv1.ResourceMemory},
+			},
+			expected: []ResourceName{ResourceMemory},
+		}, {
+			name:     "No ControlledResources specified - used default",
+			policy:   &vpa_types.ContainerResourcePolicy{},
+			expected: []ResourceName{ResourceCPU, ResourceMemory},
+		}, {
+			name:     "Nil policy - default to Auto",
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state_test.go:280]: nit: "Nil policy - use default"

File: vertical-pod-autoscaler/pkg/recommender/logic/recommender_test.go (hunk 1 above)
Comment (tweks) [reply vertical-pod-autoscaler/pkg/recommender/logic/recommender_test.go:67]: Done.

File: vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state_test.go (hunk 2 above)
Comment (tweks) [reply vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state_test.go]: Done.

**Comment by tweks** [comment]
> Looks good. Nits only.
> 
> Can you add an e2e test for this functionality as a followup?

Thanks. Yes, I will add e2e tests in next PR.

**Pull request merged**